cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -434,4 +435,36 @@ public static String getJavaSystemPropertiesToString(List<SystemProperty> javaSy
         }
         return String.join(" ", javaSystemPropertiesList);
     }
+
+    public static List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.startsWith("#") && !line.isEmpty())   {
+                validLines.add(line);
+            }
+        }
+
+        return validLines;
+    }
+
+    public static Map<String, String> configMap2Map(ConfigMap configMap, String key) {
+        Map<String, String> result = new HashMap<>();
+        if (configMap == null || configMap.getData() == null || configMap.getData().get(key) == null) {
```

Comment:
I wonder if you should separate into separate method extraction fo the key from the CM and have here only a method which goes String against String.